src/client/method_monitor.c

```diff
@@ -325,7 +325,13 @@ static Node *
                          const char *node_state,
                          uint64_t last_seen_timestamp) {
         Node *node = malloc0(sizeof(Node));
+        if (node == NULL) {
+                return NULL;
+        }
         node->connection = malloc0(sizeof(NodeConnection));
+        if (node->connection == NULL) {
+                return NULL;
```

Comment:
Shouldn't you deallocate the `node` object before leaving?

Reply:
Yes, good point. Updated it. Could you have another look? @mwperina